app/jobs/appointment_notification/worker.rb

```diff
@@ -4,27 +4,28 @@ class AppointmentNotification::Worker
 
   sidekiq_options queue: :high
 
-  DEFAULT_LOCALE = :en
-
   def metrics
     @metrics ||= Metrics.with_object(self)
   end
 
-  def perform(appointment_reminder_id, communication_type)
+  def perform(appointment_reminder_id)
     metrics.increment("attempts")
     unless Flipper.enabled?(:appointment_reminders)
       metrics.increment("skipped.feature_disabled")
       return
     end
     reminder = AppointmentReminder.includes(:appointment, :patient).find(appointment_reminder_id)
-    if reminder.appointment.previously_communicated_via?(communication_type)
+    communication_type = reminder.next_communication_type
+    unless communication_type
       metrics.increment("skipped.previously_communicated")
       return
     end
-    if reminder.status != "scheduled"
-      report_error("scheduled appointment reminder has invalid status")
+
+    unless reminder.status == "scheduled"
```

Comment:
Here is the possible race condition.  Also, this can use the question method:

```
reminder.status_scheduled?
```

Reply:
Done.